boot_node/src/config.rs

```diff
@@ -130,3 +131,26 @@ impl<T: EthSpec> TryFrom<&ArgMatches<'_>> for BootNodeConfig<T> {
         })
     }
 }
+
+/// The set of configuration parameters that can safely be (de)serialized.
+///
+/// Its fields are a subset of the fields of `BootNodeConfig`.
+#[derive(Serialize, Deserialize)]
+pub struct BootNodeConfigSerialization {
+    pub listen_socket: SocketAddr,
+    // TODO: Generalise to multiaddr
+    pub boot_nodes: Vec<Enr>,
+    pub local_enr: Enr,
+}
+
+impl BootNodeConfigSerialization {
+    /// Returns a `BootNodeConfigSerialization` obtained from copying resp. cloning the
+    /// relevant fields of `config`
+    pub fn from_config_ref<T: EthSpec>(config: &BootNodeConfig<T>) -> Self {
```

Comment:
If we adopt the following pattern, we can force a compile time error if a field is added to `BeaconNodeConfig` without also considering if it should be added to `BeaconNodeConfigSerialization`:

```rust
    pub fn from_config_ref<T: EthSpec>(config: &BootNodeConfig<T>) -> Self {
        let BootNodeConfig {
            listen_socket,
            boot_nodes,
            local_enr,
            local_key: _,
            discv5_config: _,
            phantom: _,
        } = config;

        BootNodeConfigSerialization {
            listen_socket: *listen_socket,
            boot_nodes: boot_nodes.clone(),
            local_enr: local_enr.clone(),
        }
    }
```

It's a little more verbose, but I think it would be nice to force the future developer to consider the relationship between these structs.

Reply:
Neat! Done in 71013e7.